src/webportal/src/app/components/auto-complete.jsx

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import {elementContains, Callout, TextField, DirectionalHint, getTheme, CommandButton, ColorClassNames, KeyCodes} from 'office-ui-fabric-react';
+import {isEmpty} from 'lodash';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState, useCallback, useMemo, useRef} from 'react';
+
+function getSpacingValue(unitString) {
+  const numericalPart = parseFloat(unitString);
+  if (isNaN(numericalPart)) {
+    return 0;
+  }
+  const unitPart = unitString.substring(numericalPart.toString().length).trim();
+  if (!isEmpty(unitPart) && unitPart !== 'px') {
+    throw new Error(`AutoComplete's gap only support 'px' unit`);
+  }
+  return numericalPart;
+}
+
+export const AutoComplete = ({items, value, onChange, showAllSuggestions}) => {
+  const [focused, setFocused] = useState(false);
+  const [suggestions, setSuggestions] = useState([]);
+  const [suggested, setSuggested] = useState(0);
+  const root = useRef();
+  const input = useRef();
+  // get callout's gap space
+  const {spacing} = getTheme();
+  const gap = useMemo(
+    () => getSpacingValue(spacing.s1),
+    [spacing],
+  );
+  // suggestions
+  useEffect(
```

Comment:
Can you give a simple example about how to use it?
And the auotComplete can support only autocomplete the last word? For example I have a sentence: "May the force be with you", and I input "May the fo", the suggestion should be "May the force".

Reply:
The AutoComplete component behaves like a combo box. So you need to provide the whole sentence "May the force".
Since the value and suggestionItems are controlled by its parent, you can add your own suggestion logic.
e.g. add "May the force" option only when user has input "May the f"

Reply:
If you want a multiline word/phrase level AutoComplete component, I recommend you use Monaco Editor.

Reply:
Got it, thanks for the explanation.